src/sentry/eventstream/kafka/backend.py

```diff
@@ -104,13 +105,46 @@ def strip_none_values(value: Mapping[str, Optional[str]]) -> Mapping[str, str]:
                 ),
             }
 
+    @staticmethod
+    def _is_transaction_event(event):
+        return event.group_id is None
+
+    def insert(
+        self,
+        group,
+        event,
+        is_new,
+        is_regression,
+        is_new_group_environment,
+        primary_hash,
+        received_timestamp,  # type: float
+        skip_consume=False,
+    ):
+        message_type = "transaction" if self._is_transaction_event(event) else "error"
+        assign_partitions_randomly = killswitch_matches_context(
+            "kafka.send-project-events-to-random-partitions",
+            {"project_id": event.project_id, "message_type": message_type},
+        )
+        return super().insert(
+            group,
+            event,
+            is_new,
+            is_regression,
+            is_new_group_environment,
+            primary_hash,
+            received_timestamp,
+            skip_consume,
+            assign_partitions_randomly,
+        )
+
     def _send(
         self,
         project_id: int,
         _type: str,
         extra_data: Tuple[Any, ...] = (),
         asynchronous: bool = True,
         headers: Optional[Mapping[str, str]] = None,
+        assign_partition_randomly: bool = False,
```

Comment:
Nit: I feel it would be better if you call this `skip_semantic_partitioning` or something along those lines. The purpose of the variable becomes more clear since its related to kafka.

Reply:
sure, sounds like an improvement.